safe-global/web-core
# refactor: delay switching networks until execution

## What it solves

Resolves #1758 

So far the user wallet always had to be on the same chain as the Safe in order to interact with many parts of the UI. This removes that limitation and instead shows a warning message, requesting chain switch when submitting.

- Globally initialises the Core SDK in read-only mode instead of with a signer
- Prompts the wallet to switch to the correct network before connecting to the Core SDK instance
- Removes `useTxSender` as it is not needed anymore

Note: Safe creation has not been touched.

## Other changes

- Use `safe.chainId` instead of `useChainId` when executing Safe App/spending limit txs

## How to test it

The key areas that need testing are transaction related but this likely needs a full regression. This needs to be tested with hardware wallets, WC, Mobile pairing and MM.

1. Open the Safe
2. Connect your wallet on a different chain than the safe
3. Observe there are no limitations in the UI (apart from the "Switch to" buttons in the header/account centre)
4. Create a transaction
5. Observe the new warning
6. Clicking Submit first requests you to switch chain

Note: this has been tested with WC, pairing, MM, Trezor Model T, Ledger Nano X successfully.

## Screenshots

![image](https://user-images.githubusercontent.com/20442784/227200657-9597395b-2772-4933-a83b-135c0dafc1d3.png)

## Checklist
